src/entities/postgres-storage/db-shelley.js

```diff
@@ -45,7 +47,53 @@ class DBShelley extends DB {
   async storeTx(tx: ShelleyTxType,
     txUtxos:Array<mixed> = [], upsert: boolean = true): Promise<void> {
     const { certificate } = tx
-    await super.storeTx(tx, txUtxos, upsert)
+    await super.storeTxImpl(tx, txUtxos, upsert, (txDbData: TxDbDataType): object => {
+      const wasm = global.jschainlibs
+      const {
+        inputAddresses, outputAddresses,
+      } = txDbData
+      const allAddresses = _.uniq([...inputAddresses, ...outputAddresses])
+      console.log(`metadataCreator.allAddresses = ${allAddresses}`)
+      return allAddresses.map(addressString => {
+        let address
+        try {
+          address = wasm.Address.from_bytes(Buffer.from(addressString, 'hex'))
+        } catch (e) {
+          const prefix = addressString.substring(0, 3)
+          // TODO: find a better way to distinguish legacy funds?
+          if (prefix != 'Ddz' && prefix != 'Ae2') {
```

Comment:
This will fail on testnet addresses. You can see how I do it in Yoroi-Frontend here https://github.com/Emurgo/yoroi-frontend/blob/shelley/app/api/ada/lib/storage/bridge/utils.js

Reply:
We can change this later as we shouldn't be running into legacy testnet addresses in the jormungandr deployment in `master-shelley`, as the byron-era initial funds should all be mainnet.